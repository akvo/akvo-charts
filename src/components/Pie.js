import React from 'react';
import { useECharts } from '../hooks';

const getOptions = ({ config = {}, data = [] }) => ({
  ...config,
  series: [
    {
      type: 'pie',
      data: data.map((item) => ({ value: item.value, name: item.label }))
    }
  ]
});

const Pie = ({ config, data }) => {
  const chartRef = useECharts({ config, data, getOptions });

  return (
    <div
      ref={chartRef}
      role="figure"
      style={{ width: '100%', height: '400px' }}
    />
  );
};

export default Pie;
