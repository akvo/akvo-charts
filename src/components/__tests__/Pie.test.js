import React from 'react';
import { render, screen } from '@testing-library/react';
import Pie from '../Pie';
import renderer from 'react-test-renderer';

test('renders Pie component', () => {
  const data = [
    { label: 'Category A', value: 30 },
    { label: 'Category B', value: 70 }
  ];

  const config = {
    title: 'Pie Chart Example'
  };

  render(
    <Pie
      config={config}
      data={data}
    />
  );

  const chartContainer = screen.getByRole('figure');
  expect(chartContainer).toBeInTheDocument();
});

test('renders Donut component', () => {
  const data = [
    { label: 'Category A', value: 30 },
    { label: 'Category B', value: 70 }
  ];

  const config = {
    title: 'Donut Chart Example'
  };

  render(
    <Pie
      config={config}
      data={data}
      type="donut"
    />
  );

  const chartContainer = screen.getByRole('figure');
  expect(chartContainer).toBeInTheDocument();
});

test('matches Pie snapshot', () => {
  const data = [
    { label: 'Category A', value: 30 },
    { label: 'Category B', value: 70 }
  ];

  const config = {
    title: 'Pie Chart Example'
  };

  const tree = renderer
    .create(
      <Pie
        config={config}
        data={data}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches Donut snapshot', () => {
  const data = [
    { label: 'Category A', value: 30 },
    { label: 'Category B', value: 70 }
  ];

  const config = {
    title: 'Donut Chart Example'
  };

  const tree = renderer
    .create(
      <Pie
        config={config}
        data={data}
        type="donut"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
