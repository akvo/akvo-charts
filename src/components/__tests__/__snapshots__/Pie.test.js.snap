// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Donut snapshot 1`] = `
<div
  className="container"
  role="figure"
/>
`;

exports[`matches Pie snapshot 1`] = `
<div
  className="container"
  role="figure"
/>
`;
